roles/replace_node/tasks/volume.yml

```diff
@@ -17,36 +17,15 @@
       with_items: "{{ volumes }}"
       register: brick_list
 
-    - set_fact:
-        volume_bricks: {}
-
-    - name: Get the list of bricks into a variable
-      set_fact:
-        volume_bricks: "{{ volume_bricks|combine({item.item: item.stdout}) }}"
-      loop: "{{ brick_list.results }}"
   delegate_to: "{{ gluster_maintenance_cluster_node }}"
   
 - name: Run replace-brick commit on the brick
   shell: >
-    gluster volume replace-brick {{ item.key }}
-    {{gluster_maintenance_old_node}}:{{item.value}}
-    {{gluster_maintenance_new_node}}:{{item.value}}
+    gluster volume replace-brick {{ item.0.item }}
+    {{gluster_maintenance_old_node}}:{{item.1}}
+    {{gluster_maintenance_new_node}}:{{item.1}}
     commit force
-  loop: "{{ lookup('dict', volume_bricks) }}"
+  with_subelements: 
```

Comment:
isn't the approach syntactically the same here compared to the previous implementation?

Reply:
Previously, from the results variable, new dictionary is created, with key as volume name and the brick path as value. But that doesn't provided the right way of to access values, if there are multiple values for brick paths. I removed that dictionary and directly using the results ( registered variable ) to access key and values. This approach provides access to multiple values in dictionary values